### ISSUE TITLE
retrieving name of service in run script

### ISSUE THREAD

**Issue by rbro** [just-containers/s6-overlay#447]
First, thank you for creating s6-overlay.  I'm finding it extremely useful for a nginx/php-fpm application where I need to have multiple processes running in a container.

I listed more background below, but my question is: in a s6-rc run script, is there a way to get the name of the service, similar to $SUPERVISOR_PROCESS_NAME that supervisord has?

I'm hardcoding the name of the service name in the run script for now.  I could also get the basename of the directory that the run script is in, but I wasn't sure if there was another way.

The background for my question is I'd like the logs from the supervised services go to stdout/stderr, but I prefer if each line was prefixed such as [nginx] or [php-fpm], so that I can filter/route them appropriately.

To handle it in a run script, I found:

https://serverfault.com/questions/754970/how-to-add-a-prefix-label-for-services-to-supervisor-stdout-output/946271#946271

This is working great with the slight downside that I have to hardcode the service name in each run script.

Thanks for your help.

**Comment by skarnet** [comment]
No, there is no introspection feature at all in s6 or s6-rc (although just the service name could be a simple thing to add in a future version). Your idea is correct, and something like `basename $(s6-linkname -f .)` is probably the best way to get the name of your service as defined in the s6-rc database.

**Comment by rbro** [comment]
Thanks, I've been exploring how to use s6 for logging in a serverless environment such as AWS Fargate, where I don't necessarily have access to the host's filesystem and each container is temporary and could disappear at any time.  The logs will be routed to an external system - possibly CloudWatch or Elasticsearch.

One path I was looking at was from the 12-factor app list where it says that all logs should log to stdout - https://12factor.net/logs - so I was trying to prefix the logs.  I would then use a docker logging driver to collect the logs and route them.

Another approach I was considering was to run fluentd as another s6-rc process inside my container and route the logs with that, not going through stdout at all.

Could the dedicated loggers in s6 be used in this type of environment?

Thanks again.

**Comment by skarnet** [comment]
I'm not at all knowledgeable about these tools. If they're able to process several logging channels at once, then you can use dedicated loggers for all your services that feed logs into the channels you have defined. Or, if they're able to process automatically rotated log directories with archived files, you can use s6-overlay's `logutil-service` to log into a directory and pass that directory to the tool. Otherwise, if they can only read one stream of input, then dedicated loggers aren't of any help to you and you can just let everything fall through to stdout - you don't even want to use a catch-all logger if all you need is one stream of logs to the container's stdout.

**Comment by skarnet** [comment]
Closing this.

**Comment by felipecrs** [comment]
I'm also here for log prefixing. I have logutil-service running for all my services, copying back logs to stdout `S6_LOGGING_SCRIPT="T 1 n1 s10000000 T"`. I'd like to prefix the logs that go to stdout with the service name, or or anything that can help disambiguate between multiple services.

I looked at the s6-log documentation and I could not find anything useful for this. By any chance do you have any idea?
